### PR TITLE
feat: update response status

### PR DESCRIPTION
**Pull request by acdibble**

### CODE REVIEW

File: packages/swap/prisma/migrations/20240208091109_remove_unused_fields/migration.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/*
+  Warnings:
+
+  - The values [EgressAmountZero] on the enum `FailedSwapReason` will be removed. If these variants are still used in the database, this will fail.
+  - You are about to drop the column `type` on the `FailedSwap` table. All the data in the column will be lost.
```
Comment (acdibble) [inline packages/swap/prisma/migrations/20240208091109_remove_unused_fields/migration.sql:5]: I don't think this was ever released so we an ignore this warning

File: packages/swap/prisma/migrations/20240208092708_add_failed_event_info/migration.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+-- AlterTable
+ALTER TABLE "public"."FailedSwap" ADD COLUMN     "failedAt" TIMESTAMP(3),
+ADD COLUMN     "failedBlockIndex" TEXT;
+
+-- backfill with default values
+UPDATE "public"."FailedSwap"
+SET "failedAt" = 'epoch', "failedBlockIndex" = '0-0';
```
Comment (acdibble) [inline packages/swap/prisma/migrations/20240208092708_add_failed_event_info/migration.sql:7]: Backfill existing failed swaps with dummy data

File: packages/swap/src/event-handlers/depositIgnoredV120.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        .transform((o) => {
+          try {
+            return encodeBitcoinAddress(o.value, env.CHAINFLIP_NETWORK);
+          } catch {
+            return null;
+          }
+        }),
+      hexString,
+    ]),
+    reason: reasonSchema,
```
Comment (acdibble) [inline packages/swap/src/event-handlers/depositIgnoredV120.ts:36]: v120 adds `reason`. I initially added this to the original handler with a default reason, but then I realized that if this field is removed, we wouldn't catch that error and would silently use the default value which may be wrong.

File: packages/swap/prisma/schema.prisma [hunk 4]
```diff
@@ -137,32 +137,25 @@ model SwapFee {
...
+  reason               FailedSwapReason
   destAddress          String
   destChain            Chain
   depositAmount        Decimal             @db.Decimal(30, 0)
   srcChain             Chain
   swapDepositChannel   SwapDepositChannel? @relation(fields: [swapDepositChannelId], references: [id])
   swapDepositChannelId BigInt?
   txHash               String?
+  failedAt             DateTime
+  failedBlockIndex     String
```
Comment (acdibble) [inline packages/swap/prisma/schema.prisma:159]: save some extra info for the end users

File: packages/sdk/src/swap/types.ts [hunk 5]
```diff
@@ -54,16 +54,20 @@ export interface SwapStatusRequest {
   id: string;
 }
 
-export interface CommonStatusFields extends ChainsAndAssets {
+export interface CommonStatusFields {
+  srcAsset: Asset;
+  srcChain: Chain;
+  destAsset: Asset | undefined;
+  destChain: Chain | undefined;
```
Comment (acdibble) [inline packages/sdk/src/swap/types.ts:61]: failed swaps with a vault origin won't have these fields

File: packages/swap/prisma/schema.prisma [hunk 6]
```diff
@@ -137,32 +137,26 @@ model SwapFee {
...
 model FailedSwap {
   id                   Int                 @id @default(autoincrement())
-  type                 FailedSwapType
-  reason               FailedSwapReason?
+  reason               FailedSwapReason
   destAddress          String
   destChain            Chain
   depositAmount        Decimal             @db.Decimal(30, 0)
   srcChain             Chain
+  srcAsset             Asset
```
Comment (acdibble) [inline packages/swap/prisma/schema.prisma:154]: need this for the vault origin

File: packages/sdk/src/swap/types.ts (hunk 5 above)
Comment (niklasnatter) [reply packages/sdk/src/swap/types.ts:61]: sad

**Review by niklasnatter — APPROVED** [review]

**Review by jerryafr — APPROVED** [review]

**Pull request merged**